projects/zenvdb/VDBRenormalize.cpp

```diff
@@ -51,31 +51,40 @@ static int defVDBRenormalizeSDF = zeno::defNodeClass<VDBRenormalizeSDF>("VDBReno
      "openvdb",
      }});
 
-struct  VDBSmoothSDF : zeno::INode {
-  virtual void apply() override {
-
-    auto inoutSDF = get_input("inoutSDF")->as<VDBFloatGrid>();
-    int width = std::get<int>(get_param("width"));
-    int iterations = std::get<int>(get_param("iterations"));
-    auto lsf = openvdb::tools::Filter<openvdb::FloatGrid>(*(inoutSDF->m_grid));
-    lsf.setGrainSize(1);
-    lsf.gaussian(width, iterations, nullptr);
-    //openvdb::tools::ttls_internal::smoothLevelSet(*inoutSDF->m_grid, normIter, halfWidth);
-    set_output("inoutSDF", get_input("inoutSDF"));
-  }
+struct VDBSmooth : zeno::INode {
+    virtual void apply() override {
+        auto inoutVDBtype = get_input<VDBGrid>("inoutVDB")->getType();
+        int width = get_input<NumericObject>("width")->get<int>();
+        int iterations = get_input<NumericObject>("iterations")->get<int>();
+        if (inoutVDBtype == std::string("FloatGrid")) {
+            auto inoutVDB = get_input("inoutVDB")->as<VDBFloatGrid>();
+            auto lsf = openvdb::tools::Filter<openvdb::FloatGrid>(*(inoutVDB->m_grid));
+            lsf.setGrainSize(1);
+            lsf.gaussian(width, iterations, nullptr);
+            //openvdb::tools::ttls_internal::smoothLevelSet(*inoutSDF->m_grid, normIter, halfWidth);
+            set_output("inoutVDB", get_input("inoutVDB"));
+        }
+        else if (inoutVDBtype == std::string("Vec3fGrid")) {
+            auto inoutVDB = get_input("inoutVDB")->as<VDBFloat3Grid>();
+            auto lsf = openvdb::tools::Filter<openvdb::Vec3fGrid>(*(inoutVDB->m_grid));
+            lsf.setGrainSize(1);
+            lsf.gaussian(width, iterations, nullptr);
+            set_output("inoutVDB", get_input("inoutVDB"));
+        }
+    }
 };
 
-static int defVDBSmoothSDF = zeno::defNodeClass<VDBSmoothSDF>("VDBSmoothSDF",
-     { /* inputs: */ {
-     "inoutSDF", 
-     }, /* outputs: */ {
-     "inoutSDF",
-     }, /* params: */ {
-         {"int", "width", "1"},
-         {"int", "iterations", "1"},
-     }, /* category: */ {
-     "openvdb",
-     }});
+ZENO_DEFNODE(VDBSmooth)(
+    { /* inputs: */ {
+    "inoutVDB",
+    {"int", "width", "1"},
+    {"int", "iterations", "1"},
```

Comment:
Still use params... or maybe you could use a copy of this node instead to prevent breaking old graphs?